test/execution/compiler_test.cpp

```diff
@@ -724,28 +794,29 @@ TEST_F(CompilerTest, SimpleAggregateTest) {
   auto exec_ctx = MakeExecCtx(std::move(callback), agg->GetOutputSchema().Get());
 
   // Run & Check
-  auto executable = ExecutableQuery(common::ManagedPointer(agg), common::ManagedPointer(exec_ctx));
-  executable.Run(common::ManagedPointer(exec_ctx), MODE);
+  auto executable =
+      execution::compiler::CompilationContext::Compile(*agg, exec_ctx->GetExecutionSettings(), exec_ctx->GetAccessor());
+  executable->Run(common::ManagedPointer(exec_ctx), MODE);
   multi_checker.CheckCorrectness();
 
   // Pipeline Units
-  auto pipeline = executable.GetPipelineOperatingUnits();
+  auto pipeline = executable->GetPipelineOperatingUnits();
   EXPECT_EQ(pipeline->units_.size(), 2);
 
-  auto feature_vec0 = pipeline->GetPipelineFeatures(execution::pipeline_id_t(0));
-  auto feature_vec1 = pipeline->GetPipelineFeatures(execution::pipeline_id_t(1));
-  auto exp_vec0 = std::vector<brain::ExecutionOperatingUnitType>{
-      brain::ExecutionOperatingUnitType::AGGREGATE_BUILD, brain::ExecutionOperatingUnitType::SEQ_SCAN,
-      brain::ExecutionOperatingUnitType::OP_INTEGER_COMPARE,
-      brain::ExecutionOperatingUnitType::OP_INTEGER_PLUS_OR_MINUS};
-  auto exp_vec1 = std::vector<brain::ExecutionOperatingUnitType>{brain::ExecutionOperatingUnitType::AGGREGATE_ITERATE,
+  auto feature_vec0 = pipeline->GetPipelineFeatures(execution::pipeline_id_t(1));
```

Comment:
Two things: (1) the first feature of the iterate side should be `AGGREGATE_ITERATE` instead of `HASH_AGGREGATE` because it's the iterate side. (2) I'm not sure why the iterate side is the first pipeline and the build side is the second pipeline. Does the new compiler generate pipelines in reverse order? Seems counterintuitive for me.